Mortar method rotation axis from z to x, y or z (#2016)
chaos-polymtl/lethe: release_notes/current/2026_07_22_Tremblay.md

## [Master] - 2026/07/22

### Added

> - MAJOR Implements the ability to use a different rotation axis than the z-axis when using the mortar method. Also makes it so that the rotation axis doesn't need to pass through the origin. The code is still limited to a rotation axis parallel to the main x, y or z axis. [#2016](https://github.com/chaos-polymtl/lethe/pull/2016)

Reviewer: ```suggestion
- MAJOR Implements the ability to use a different rotation axis than the z-axis when using the mortar method. The code is still limited to a rotation axis parallel to the main x, y or z axis. [#2016](https://github.com/chaos-polymtl/lethe/pull/2016)
```
We were prescribing the center of rotation before, so we could already have a problem centered outside the origin

Author: True, I moved it to "Fixed" but forgot to remove it from there

### Fixed

> - MINOR Fixes a bug when calculating the normal vectors at the mortar interface when the rotation axis is not passing through the origin. The code would still assume the center of rotation to be at (0,0), which would affect the results. [#2016](https://github.com/chaos-polymtl/lethe/pull/2016)

Reviewer 2: ```suggestion
- MINOR Fixes a bug when calculating the normal vectors at the mortar interface when the rotation axis does not pass through the origin. The previous code would assume the center of rotation to be at (0,0) regardless, which would affect the results. [#2016](https://github.com/chaos-polymtl/lethe/pull/2016)
```

> - MINOR Fixes a bug when calculating the normal vectors at the mortar interface when the rotation axis is not passing through the origin. The code would still assume the center of rotation to be at (0,0), which would affect the results. [#2016](https://github.com/chaos-polymtl/lethe/pull/2016)

Reviewer 2: So does this PR fix non-origin centered axes (in both 2D and 3D)? If so I don't think we test for that yet in 3D.

Author: It does, and yes we could modify an existing 3D test by shifting the rotation axis, we only test the feature in 2D right now.
